src/core/coap/coap.cpp

```diff
@@ -654,6 +654,9 @@ void CoapBase::ProcessReceivedResponse(Message &aMessage, const Ip6::MessageInfo
     case kTypeConfirmable:
         // Send empty ACK if it is a CON message.
         IgnoreError(SendAck(aMessage, aMessageInfo));
+#ifdef __ANDROID__
+        [[clang::fallthrough]];
+#endif
```

Comment:
Is it possible to suppress the clang fall-through warning?

Reply:
Yes, we can add `-Wno-implicit-fallthrough` to cppflags to suppress the clang fall-through warning. Should we suppress this warning? @jwhui @bukepo

Reply:
Not feeling good to have the platform-specific code everywhere :)

Another option is to have a macro `FALL_THROUGH` in `openthread/toolchain.h`:
```c++
#if __cplusplus >= 201703L
#define FALL_THROUGH    [[fallthrough]]
#if __CLANG__
#define FALL_THROUGH    [[clang::fallthrough]]
#elif __has_attribute(__fallthrough__)
# define FALL_THROUGH    __attribute__((__fallthrough__))
#else
# define FALL_THROUGH    do {} while (0)  /* fallthrough */
#endif
```

See https://elixir.bootlin.com/linux/v5.10/source/include/linux/compiler_attributes.h#L207

Reply:
Will try it.

Reply:
Agree that we should avoid the `#if` statements.

Recognizing `// fall through` is common for existing toolchains, including gcc and coverity.

Reply:
Updated